### PR TITLE
Lorawan rtc

### PR DESCRIPTION
**Pull request by FRASTM**
stm32 BSP LoRaWan with RTC instance

### CODE REVIEW

File: src/BSP/timer_if.c [hunk 1]
```diff
@@ -223,22 +253,15 @@ UTIL_TIMER_Status_t TIMER_IF_StartTimer(uint32_t timeout)
...
-  sAlarm.AlarmMask = RTC_ALARMMASK_NONE;
-  sAlarm.AlarmSubSecondMask = RTC_ALARMSUBSECONDBINMASK_NONE;
-  sAlarm.Alarm = RTC_ALARM_A;
-  if (HAL_RTC_SetAlarm_IT(&hrtc, &sAlarm, RTC_FORMAT_BCD) != HAL_OK)
-  {
-    Error_Handler();
-  }
+
+  /* Program ALARM A on subsecond, mask is 32 (and fixed to RTC_ALARMMASK_NONE for calendar) */
+  RTC_StartAlarm(RTC_ALARM_A, 0, 0, 0, 0, UINT32_MAX - timeout, RTC_HOURFORMAT12_PM, 32UL);
```
Comment (fpistm) [inline src/BSP/timer_if.c]: Use RTC_ALARM_B instead to reserve RTC_ALARM_A for other usage.

File: src/BSP/timer_if.c [hunk 2]
```diff
@@ -251,10 +274,15 @@ UTIL_TIMER_Status_t TIMER_IF_StopTimer(void)
   /* USER CODE BEGIN TIMER_IF_StopTimer */
 
   /* USER CODE END TIMER_IF_StopTimer */
+#if 1
   /* Clear RTC Alarm Flag */
   __HAL_RTC_ALARM_CLEAR_FLAG(&hrtc, RTC_FLAG_ALRAF);
   /* Disable the Alarm A interrupt */
   HAL_RTC_DeactivateAlarm(&hrtc, RTC_ALARM_A);
```
Comment (fpistm) [inline src/BSP/timer_if.c]: ditto
RTC_ALARM_B

File: src/BSP/timer_if.c [hunk 3]
```diff
@@ -251,10 +274,15 @@ UTIL_TIMER_Status_t TIMER_IF_StopTimer(void)
...
+#else
+  rtc.disableAlarm(RTC_ALARM_A);
```
Comment (fpistm) [inline src/BSP/timer_if.c]: ditto 
RTC_ALARM_B

File: src/BSP/timer_if.c (hunk 1 above)
Comment (fpistm) [inline src/BSP/timer_if.c]: why RTC_HOURFORMAT12_PM ? 
below it is set to 
rtc.begin(true, STM32RTC::HOUR_24);

**Review by fpistm — CHANGES_REQUESTED** [review]

**Comment by FRASTM** [comment]
Using the **ALARM B** for the LoraWan and keep Alarm A for other purpose
Remove `#if 0/1` code  sections

File: src/BSP/timer_if.c (hunk 1 above)
Comment (FRASTM) [reply src/BSP/timer_if.c]: yes, the parameter in the RTC_StartAlarm is of type `hourAM_PM_t` (HOUR_AM or  HOUR_PM) which will be ignored in the function :  ` /* Ignore time AM PM configuration if in 24 hours format */`

Comment (FRASTM) [reply src/BSP/timer_if.c]: Use the RTC handle as a global from the STM32_RTC library 
Use Alarm B of the RTC to trig on SubSeconds

**Pull request closed without merging**